collector-http/src/test/java/zipkin2/collector/otel/http/ITOpenTelemetryHttpCollector.java

```diff
@@ -29,13 +88,288 @@ class ITOpenTelemetryHttpCollector {
         .storage(store)
         .build()
         .start();
+    ServerBuilder serverBuilder = Server.builder().http(port);
+    collector.reconfigure(serverBuilder);
     metrics = metrics.forTransport("otel/http");
+    server = serverBuilder.build();
+    server.start().join();
   }
 
-  @AfterEach void teardown() throws IOException {
+  @AfterEach
+  void teardown() throws IOException {
     store.close();
     collector.close();
+    server.stop().join();
   }
 
-  // TODO: integration test
+  @Test
+  void testServerKind() throws Exception {
+    List<String> traceIds = new ArrayList<>();
+    List<String> spanIds = new ArrayList<>();
+    final int size = 5;
+    for (int i = 0; i < size; i++) {
+      Span span = tracer
+          .spanBuilder("get")
+          .setSpanKind(SpanKind.SERVER)
+          .setAttribute("string", "foo" + i)
+          .setAttribute("int", 100)
+          .setAttribute("double", 10.5)
+          .setAttribute("boolean", true)
+          .setAttribute(AttributeKey.stringArrayKey("array"), Arrays.asList("a", "b", "c"))
+          .setAttribute(NetworkAttributes.NETWORK_LOCAL_ADDRESS, "127.0.0.1")
+          .setAttribute(NetworkAttributes.NETWORK_LOCAL_PORT, 12345L)
+          .startSpan();
+      Thread.sleep(100); // do something
+      span.end();
+      spanIds.add(span.getSpanContext().getSpanId());
+      traceIds.add(span.getSpanContext().getTraceId());
+    }
+    Awaitility.waitAtMost(Duration.ofSeconds(5))
+        .untilAsserted(() -> assertThat(store.acceptedSpanCount()).isEqualTo(size));
+    List<List<zipkin2.Span>> received = store.getTraces(traceIds).execute();
+    assertThat(received.size()).isEqualTo(size);
+    for (int i = 0; i < size; i++) {
+      assertThat(received.get(i)).hasSize(1);
+      zipkin2.Span span = received.get(i).get(0);
+      assertThat(span.id()).isEqualTo(spanIds.get(i));
+      assertThat(span.traceId()).isEqualTo(traceIds.get(i));
+      assertThat(span.parentId()).isNull();
+      assertThat(span.name()).isEqualTo("get");
+      assertThat(span.kind()).isEqualTo(zipkin2.Span.Kind.SERVER);
+      assertThat(span.tags()).hasSize(9);
+      assertThat(span.tags()).containsEntry("string", "foo" + i);
+      assertThat(span.tags()).containsEntry("int", "100");
+      assertThat(span.tags()).containsEntry("double", "10.5");
+      assertThat(span.tags()).containsEntry("boolean", "true");
+      assertThat(span.tags()).containsEntry("array", "[\"a\",\"b\",\"c\"]");
```

Comment:
interesting to serialize the json, but guess that's how otel does it. Before otel we just joined on comma as easier on indexes.

Reply:
While https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/README.md#attribute
says 
> For protocols that do not natively support non-string values, non-string values SHOULD be represented as JSON-encoded strings. For example, the expression int64(100) will be encoded as 100, float64(1.5) will be encoded as 1.5, and an empty array of any type will be encoded as [].

the test-case doesn't have square brackets  https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/zipkin/src/test/java/io/opentelemetry/exporter/zipkin/OtelToZipkinSpanTransformerTest.java#L382-L385

I thought `OtelToZipkinSpanTransformer` was not implemented as documented, but do you thing aligning the existing implementation is better?

Reply:
yes align with existing impl and make a comment of the source line you are referring to and that it is understood to be not the same as the otel spect. The reason is that the experience is worse to json encode, since folks have comma splitters. I suspect that may be why upstream in otel they are not json encoding.

Reply:
fixed in https://github.com/openzipkin-contrib/zipkin-otel/pull/13/commits/420507baaf023126a889d6ac142dff18256b7491